torch/csrc/inductor/aoti_torch/tensor_converter.h

```diff
@@ -16,17 +16,45 @@ TORCH_API at::Tensor* tensor_handle_to_tensor_pointer(AtenTensorHandle handle);
 // No ownership transfer, just pointer type conversion
 TORCH_API AtenTensorHandle tensor_pointer_to_tensor_handle(at::Tensor* tensor);
 
-// unsafe_alloc_new_handles_from_tensors is used for allocating new aten
-// tensor objects and return them as a vector of AtenTensorHandle (raw
-// pointers), and those pointers will be stolen by model.so.
-TORCH_API std::vector<AtenTensorHandle> unsafe_alloc_new_handles_from_tensors(
-    std::vector<at::Tensor>& tensors);
-
 // alloc_tensors_by_stealing_from_handles is used for creating a vector of aten
 // tensors by stealing from a vector of handles
 // WARNING: only used in the non ABI compatible mode
 TORCH_API std::vector<at::Tensor> alloc_tensors_by_stealing_from_handles(
     std::vector<AtenTensorHandle>& handles);
 
+class TORCH_API RAIITensorToHandleAllocator {
+ public:
+  RAIITensorToHandleAllocator() = delete;
+  RAIITensorToHandleAllocator(const RAIITensorToHandleAllocator& other) =
+      delete;
+  RAIITensorToHandleAllocator(RAIITensorToHandleAllocator&& other) = delete;
+  RAIITensorToHandleAllocator& operator=(
+      const RAIITensorToHandleAllocator& other) = delete;
+  RAIITensorToHandleAllocator& operator=(RAIITensorToHandleAllocator&& other) =
+      delete;
+
+  // Allocate new tensor objects and allocate an array to store pointers
+  // (AtenTensorHandle) to those objects
+  RAIITensorToHandleAllocator(std::vector<at::Tensor>& tensors);
+
+  ~RAIITensorToHandleAllocator() {
+    if (handles_) {
+      delete[] handles_;
```

Comment:
so... who is taking care of disposing the AtenTensorHandles themselves?!

Reply:
This is what I missed, but crossing ABI is annoying. I will just drop this PR.